library/unwind/src/lib.rs

```diff
@@ -57,29 +59,30 @@ cfg_if::cfg_if! {
 // When building with crt-static, we get `gcc_eh` from the `libc` crate, since
 // glibc needs it, and needs it listed later on the linker command line. We
 // don't want to duplicate it here.
-#[cfg(all(
-    target_os = "linux",
-    target_env = "gnu",
-    not(feature = "llvm-libunwind"),
-    not(feature = "system-llvm-libunwind")
-))]
-#[link(name = "gcc_s", cfg(not(target_feature = "crt-static")))]
-extern "C" {}
-
-#[cfg(all(
-    target_os = "linux",
-    target_env = "gnu",
-    not(feature = "llvm-libunwind"),
-    feature = "system-llvm-libunwind"
-))]
-#[link(name = "unwind", cfg(not(target_feature = "crt-static")))]
-extern "C" {}
+#[cfg(all(target_os = "linux", target_env = "gnu"))]
+cfg_if::cfg_if! {
+    if #[cfg(all(feature = "llvm-libunwind", feature = "system-llvm-libunwind"))] {
+        compile_error!("`llvm-libunwind` and `system-llvm-libunwind` cannot be enabled at the same time");
+    } else if #[cfg(feature = "llvm-libunwind")] {
+        #[link(name = "unwind", kind = "static", modifiers = "-bundle", cfg(not(target_feature = "crt-static")))]
+        extern "C" {}
+    } else if #[cfg(feature = "system-llvm-libunwind")]{
+        #[link(name = "unwind", cfg(not(target_feature = "crt-static")))]
```

Comment:
Don't this also need's an `extern "C"` declaration ?
```suggestion
        #[link(name = "unwind", cfg(not(target_feature = "crt-static")))]
        extern "C" {}
```

Reply:
fixed